aarch64-unknown-none.ld

```diff
@@ -6,7 +6,7 @@ ENTRY(ram64_start)
     ACPI:   [0x4020_0000-0x403f_ffff)
     kernel: [0x4048_0000-]
    The stack start is at the end of the DRAM region. */
-ram_min = 0x40480000;
+ram_min = 0x40600000;
 
 SECTIONS
 {
```

Comment:
I don't understand how this gives more room for GRUB. This is file determines the layout of the ELF executable that is compiled from the project. Your commit message seems to imply that the EFI payload is loaded into the memory just below the RHF executable code. I think this is a bit different to how we do it elsewhere. If you want to keep it the same it would be better to.

Make `ram_min = 0x4000000` the true start of RAM and then add a `payload_start/payload_end` region to the linker file to save the memory and push back the code region.

@retrage Perhaps you can clarify on this?

Reply:
Hi @rbradford -, In current design, 0x4040000 ~ 0x4048000 is reserved for RHF payload. 0x40000000 ~0x40400000 is reserved for DTB and ACPI.

Reply:
So what does that have to do with the EFI payload?

Reply:
I think it would be better to load the kernel/grub so that it doesn't overwrite the memory that RHF is loaded at e.g. on x86 RHF is loaded at 1MiB and the EFI binary/kernel is loaded at 2MiB. The correct place to change is probably the value provided to `StartInfo::new` for `kernel_load_addr` as that is the address that is passed to the EFI PE loader.

Reply:
I think this is a big change that impact CH, RHF and maybe also future change for EDK2 (I'd like to unify the boot process of EDK2 and RHF), so let me think it through carefully.

Reply:
The points are:
* Assume that 2MB of space is enough to load GRUB. To do this:
  * Move `ram_min` to `0x40600000`.
  * Add a run-time payload size check.
I think the change in 6735c61c8929b1697d20d76310635557b0748c44 is little bit ad hoc, but it's acceptable because it works. It's future work to implement better a loading process.

So, my suggestion is to just add comments to the memory layout assumption. See https://github.com/cloud-hypervisor/rust-hypervisor-firmware/pull/262#discussion_r1257456696.

Reply:
OK